Rpackage/R/weakestLink.R

```diff
@@ -0,0 +1,91 @@
+weakest_link <- structure(function(tree){
+  tree <- nodeapply(tree, ids = 1)[[1]]
+  
+  find_weakest_link <- function(tree){
+    ### get node id and cost of the current node
+    ter_id <- nodeids(tree, terminal = TRUE)
+    all_id <- nodeids(tree, terminal = FALSE)
+    node_id <- all_id[1]
+    node_cost <- as.numeric(nodeapply(tree, ids = node_id, info_node)[[1]][2])
+    
+    
+    ### catch numeric zero of the root
+    if(length(node_cost) == 0){
+      node_cost <- tail(compute_optimal_costs(target.mat, margin, loss),1)[[3]]
+    }
+    
+    ### leaf reached
+    if(is.terminal(tree)){
+      return(rbind(c(Inf, node_id)))
+    }
+    else{
+      
+      ### terminal leaf info
+      n <- nodeapply(tree, ids = ter_id, info_node)
+      
+      ### sum of cost of tree
+      C_Tt <- sum(matrix(unlist(n), nrow = length(n), byrow = T)[,2])
+      
+      ### current alpha value
+      current_gt <- (node_cost - C_Tt)/(length(ter_id) - 1)
+      
+      ### left right child ids and partynodes
+      kids <- kids_node(tree)
+      kid_ids <- sapply(kids, id_node)
+      left_kid_id <- kid_ids[1]
+      right_kid_id <- kid_ids[2]
+  
+      ### weakest link from left tree part
+      wlink <- find_weakest_link(kids[[1]])
+      left_min_gt <- tail(wlink)[1]
```

Comment:
It's a bit obscur to me why your woul duse tail here. What does it return (data wise)?